generic3g/ComponentSpecParser/parse_geometry_spec.F90

```diff
@@ -107,14 +107,20 @@ module function parse_geometry_spec(mapl_cfg, registry, rc) result(geometry_spec
             levels = ESMF_HConfigAsR4Seq(vertical_grid_cfg, keyString='levels' ,_RC)
             vertical_grid = FixedLevelsVerticalGrid(standard_name, levels, units)
          case('model')
-            num_levels = ESMF_HConfigAsI4(vertical_grid_cfg, keyString='num_levels', _RC)
+            standard_name = ESMF_HConfigAsString(vertical_grid_cfg, keyString='standard_name', _RC)
             units = ESMF_HConfigAsString(vertical_grid_cfg, keyString='units', _RC)
-            vertical_grid = ModelVerticalGrid(num_levels=num_levels, units=units)
-            short_name = ESMF_HConfigAsString(vertical_grid_cfg, keyString='short_name', _RC)
+            num_levels = ESMF_HConfigAsI4(vertical_grid_cfg, keyString='num_levels', _RC)
+            vertical_grid = ModelVerticalGrid(standard_name=standard_name, units=units, num_levels=num_levels)
             select type(vertical_grid)
             type is(ModelVerticalGrid)
-               call vertical_grid%add_variant(short_name=short_name)
                call vertical_grid%set_registry(registry)
+               if (standard_name == "air_pressure") then
+                  call vertical_grid%add_short_names(edge="PLE", center="PL")
+               else if (standard_name == "height") then
+                  call vertical_grid%add_short_names(edge="ZLE", center="ZL")
+               else
+                  _FAIL("unsupported standard name ["//standard_name//"]")
+               end if
```

Comment:
These cannot be hardwired here.  At least not in the long term.   How about requiring them in the yaml file when you try to specify the vertical grid that way:
```yaml
vertical_grid:
    coordinate:
        dimension: pressure
        stagger: edge
        field: PLE
    coordinate:
        dimension: height
        stagger: center
        field: ZL
...
```